tools/clang/lib/Sema/SemaHLSLDiagnoseTU.cpp

```diff
@@ -354,10 +379,14 @@ void hlsl::DiagnoseTranslationUnit(clang::Sema *self) {
     }
   }
 
-  CallGraphWithRecurseGuard callGraph;
```

Comment:
Wouldn't it be more efficient to leave the declaration outside of the for loop and reuse the callgraph?

Reply:
The visited functions need to limit to current entry so we can use the entry shader kind when available.
If reuse callgraph, the visited functions will be mixed, cannot know the shader kind anymore.